text/3550-new-range.md

```diff
@@ -0,0 +1,584 @@
+
+## Mitigation
+
+To mitigate these drawbacks, we recommend introducing and stabilizing an MVP of the new types as soon as possible, well before Edition 2024 releases (even before the implementation of the syntax feature is complete). This will give libraries time to issue updates supporting the new range types.
+
+Some users may depend on libraries that are not updated before Edition 2024. These users do not just have to accept adding explicit conversions to their code. They also have the option to stay on a prior edition.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Just implement `Copy` on the types as-is
+
+`Copy` iterators are a large footgun. It was decided to [remove `Copy` from all iterators back in 2015](https://github.com/rust-lang/rust/pull/21809), and that decision is unlikely to be reversed.
+
+That said, there are a few possibilities:
+- Sophisticated lint to catch when an iterator is problematically copied
+- Language or library feature to allow `Copy` structs to have certain non-`Copy` fields
+- Specialize `IntoIterator` on these range types and lint whenever the `Iterator` impl is used
+
+None of these approaches would resolve the following serious issues:
+- `RangeInclusive` being larger than necessary for range purposes
+- Incorrect `ExactSizeIterator` implementations
+
+## Name the new types something besides `Range`
+
+We could choose to introduce these new types with a name other than `Range`. Some alternatives that have been proposed:
+- Interval
+- Span
+- Bounds
+
+We believe that it is best to keep the `Range` naming for several reasons:
+- Existing `Range*` types that implement `Copy` and not `Iterator` that won't be touched by this change
+- Large amount of legacy educational material and code using the `Range` naming
+- It's best to match the name of the syntax (["range expressions"](https://doc.rust-lang.org/reference/expressions/range-expr.html))
+
+## Use legacy range types as the iterators for the new range types
+
+We could choose to make `new_range.into_iter()` resolve to a legacy range type. This would reduce the number of new types we need to add to the standard library.
+
+But the legacy range types have a much larger API surface than other `Iterator`s in the standard library, which typically only implement the various iterator traits and maybe have a `remainder` method. Specifically, there are no iterator types in the standard library which have public fields. Nor do any implement `PartialEq`, `Eq`, `Hash`, `Index`, or `IndexMut`.
+
+`RangeInclusive` especially must take care with equality, hashing, and indexing because it can be exhausted. By removing those impls from the iterator for it, we can prevent that misuse entirely.
```

Comment:
Would be worth noting `RangeInclusive`s [performance problems](https://github.com/rust-lang/rust/issues/45222) that can be avoided for a new `RangeInclusiveIter` IMHO.